core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -804,9 +805,11 @@ private[spark] object JsonProtocol {
       case Some(values) => values.map(accumulableInfoFromJson)
       case None => Seq.empty[AccumulableInfo]
     }
+    val partitionId = jsonOption(json \ "Partition ID").map(_.extract[Int]).getOrElse(-1)
 
     val taskInfo =
       new TaskInfo(taskId, index, attempt, launchTime, executorId, host, taskLocality, speculative)
+    taskInfo.setPartitionId(partitionId)
```

Comment:
do we need a setting other vars like gettingResultTime and finishTime seem to be accessed 
taskInfo.gettingResultTime = gettingResultTime
taskInfo.finishTime = finishTime

May be make it consistent if we need we want setters and make it private for others as well?